cpp/src/toppra/geometric_path.hpp

```diff
@@ -8,7 +8,31 @@
 #include <vector>
 
 namespace toppra {
-class GeometricPath {};
+class GeometricPath {
+public:
+  GeometricPath(){};
+
+  /**
+   * /brief Evaluate the path at given position.
+   */
+  virtual Vector eval(value_type, int order = 0);
+
+  /**
+   * /brief Evaluate the path at given positions (vector).
+   */
+  virtual Vectors eval(std::vector<value_type>, int order = 0);
```

Comment:
Any reason not to use `Vector` here ? We will have to synchronize as I used `Vector` as type for vector of times.

Reply:
Noted. Will change.

Reply:
Basically I use std::vector for both coefficients and breakpoints, so for the sake of consistency I use std::vector there. Not a very strong reason though.

Reply:
Maybe here, we could provide a default implementation that calls repeatedly
```
eval(time[i], order);
```

Reply:
> Basically I use std::vector for both coefficients and breakpoints, so for the sake of consistency I use std::vector there. Not a very strong reason though.

In some places, I use the Eigen interface, e.g. https://github.com/jmirabel/toppra/blob/34e2a3f222374ac43a93936edefcd030c2512fc2/cpp/src/toppra/solver.cpp#L15